be2-scala/src/test/scala/ch/epfl/pop/model/objects/ChannelSuite.scala

```diff
@@ -61,15 +61,15 @@ class ChannelSuite extends FunSuite with Matchers {
   test("LaoId extraction channel test") {
     val laoId = "base64_lao_id";
     def channel = Channel(Channel.ROOT_CHANNEL_PREFIX + Base64Data.encode(laoId))
-    val expected = laoId.getBytes()
+    val expected = Base64Data.encode(laoId)
     noException shouldBe thrownBy(channel)
     channel.decodeSubChannel.get should equal(expected)
   }
 
   test("Real LaoId extraction channel test") {
     val laoId = "mEKXWFCMwb";
     def channel = Channel(Channel.ROOT_CHANNEL_PREFIX + Base64Data.encode(laoId))
-    val expected = laoId.getBytes()
+    val expected = Base64Data.encode(laoId)
```

Comment:
careful, here you're taking the string `mEKXWFCMwb` and transforming it into base64.
You're not _interpreting_ that string as Base64Data. Is that what you want ? 

Alternatively, you should just be able to do this:
```suggestion
    val laoId = "mEKXWFCMwb";
    def channel = Channel(Channel.ROOT_CHANNEL_PREFIX + Base64Data(laoId))
    val expected = Base64Data(laoId)
```